enterprise/server/execution_search_service/execution_search_service.go

```diff
@@ -53,6 +59,44 @@ type ExecutionWithInvocationId struct {
 	invocationID string
 }
 
+func (s *ExecutionSearchService) FetchExecutionRequestMetadata(ctx context.Context, invocationID string, execIDs []string) (map[string]*repb.RequestMetadata, error) {
+	if !*enableFetchExecutionRequestMetadata {
+		return make(map[string]*repb.RequestMetadata), nil
+	}
+	if s.oh == nil {
+		return nil, status.UnavailableError("OLAP DB is required to search execution request metadata")
+	}
+	u, err := s.env.GetAuthenticator().AuthenticatedUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if u.GetGroupID() == "" {
+		return nil, status.InvalidArgumentError("failed to find user's group when searching executions request metadata.")
+	}
+	if err := authutil.AuthorizeGroupAccess(ctx, s.env, u.GetGroupID()); err != nil {
+		return nil, err
+	}
+	q := query_builder.NewQuery(`SELECT * FROM "Executions"`)
```

Comment:
selecting fewer columns will probably reduce IO a bunch and speed up the query:

```suggestion
	q := query_builder.NewQuery(`SELECT execution_id, target_label, action_mnemonic FROM "Executions"`)
```

Reply:
great suggestion. applied.